api.js

```diff
@@ -11,63 +14,73 @@ function sendError(res, statusCode, code, description) {
 
 }
 
-app.get('/blobs/:blob_id', (req, res) => {
+app.get('/blobs/:blobId', (req, res) => {
     db.query('select content_type, blob_value from blobs where blob_id = ?',
-        [req.params.blob_id],
+        [req.params.blobId],
         (err, data) => {
             if (data.length != 1) {
                 sendError(res, 404, 'NotFound', 'Blob not found');
             } else {
-                res.contentType(data[0].content_type).send(data[0].blob_value);
+                const row = data[0];
+                res.set('Content-Type', row.content_type);
+                res.send(row.blob_value);
             }
     });
 });
 
-app.post('/contracts/:contract_id/cfdi', (req, res) => {
-    let { contractFile } = req.body;
+app.post('/contracts/:contractId/cfdi', (req, res) => {
+    if (!req.is("image/*") && !req.is("application/pdf")) {
+        sendError(res, 415, "UnsupportedMediaType", "CFDI tiene formato inválido")
+    }
+    const contractId = req.params.contractId;
+    const contractFile = req.body;
+    const contentType = req.get('Content-Type');
     db.query('select * from contracts where contract_id = ?',
-        [req.params.contract_id],
-        (err, data) => {
-            if (data[0] != null)
-            {
+        [contractId],
+        (err, rows) => {
+            if (rows.length != 1) {
+                sendError(res, 404, 'NotFound', 'Contract not found');
+                return;
+            }
+            const contract = rows[0];
+            if (contract.cfdi_id !== null) {
                 sendError(res, 400, 'CFDIAlreadyExists', 'El contrato ya tiene CFDI');
             } else {
                 db.query("insert into blobs (content_type, content) (?, ?)",
-                [req.headers['Content-Type'], contractFile],
-                (err, blobData) => {
+                [contentType, contractFile],
+                (err, results) => {
                     if (err) {
                         sendError(res, 500, 'InternalServerError', 'Could not insert');
                     } else {
+                        const blobId = results.insertId;

```

Comment:
El id del blob viene en esta variable magica del driver de mysql (lo averigue [aqui](https://github.com/mysqljs/mysql#getting-the-id-of-an-inserted-row))